plugins/datastreamer_plugin/ui/topic_selection_dialog/dialogselecttopics.cpp

```diff
@@ -282,15 +314,21 @@ void DialogSelectTopics::reset_to_configuration_()
 
     // FastDDS configuration
     // Set current domain
-    ui->domainid_spin->setValue(configuration_.domain_id_selected);
+    domain_id_selected_ = configuration_.domain_id;
+
+    ui->domainid_spin->setValue(domain_id_selected_);
     ui->current_domain_label->setText(
-        QString::number(configuration_.domain_id_connected));
+        QString::number(domain_id_connected_));
     check_domain_button_must_be_enable_();
 
     // Add XML files
+    // First erase every data in the list
+    ui->xml_files_list->clear();
     for (const auto& file : configuration_.xml_datatypes_files)
     {
-        // TODO
+        // Add every xml file in configuration
+        const auto item = new QListWidgetItem(file);
+        ui->xml_files_list->addItem(item);
```

Comment:
It is not said in the documentation: https://doc.qt.io/qt-5/qlistwidget.html#addItem and it is used like that in every example (also the ones provided by Qt). I didn't want to say it, but I do not know if Qt can assure there are no leak memory :sweat_smile: 
However, there is other API to add an item from a QString instead of an item, so `new` is not required in this case.

Reply:
ok, much better